src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +832,46 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        private void EmbedCompilerFlags(CommonPEModuleBuilder module)
+        {
+            byte[] bytes = new byte[0];
+
+            // TODO: Where to get compiler version? 
+
+            // TODO: Where to get source encoding
+
+            _debugMetadataOpt.AddCustomDebugInformation(
+                parent: EntityHandle.ModuleDefinition,
+                kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.DeterministicCompilerFlags),
+                value: _debugMetadataOpt.GetOrAddBlob(bytes));
+        }
+
+        private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
+        {
+            var bytes = new List<byte>();
+
+            // TODO: Get metadata reference information
+            // File name: Foo.exe
+            // COFF header Timestamp field (4 bytes): 0x542d5742
+            // COFF header SizeOfImage field (4 bytes): 0x32000000
+            // MVID (Guid, 24 bytes): 0x24a44d8218894463807674caf3b1c19a
+            // Output: Foo.exe\0542d5742320000000024a44d8218894463807674caf3b1c19a
+            foreach (var metadataReference in module.CommonCompilation.ExternalReferences)
```

Comment:
>ExternalReferences [](start = 71, length = 18)

I think ultimately we would want to capture all metadata references, not just the external ones. 
For now I think using ExternalReferences is fine since anything beyond that only affects scripting scenarios which are not really that important in this context.

If we need to add references resolved from #r directives in future we can create another CDI blob. They might need a bit more information anyways (such as which #r do they correspond to).

Reply:
I added a note about this in the spec document.